src/walker.cxx

```diff
@@ -0,0 +1,2531 @@
+{
+  elastic_ = emodel;
+  return 0;
+}
+
+int WalkerKremplSwitchRule::kappa(const double * const edot, double T, double &
+                                  kap)
+{
+  double edev[6];
+  std::copy(edot, edot+6, edev);
+  dev_vec(edev);
+
+  double de = std::sqrt(2.0/3.0) * norm2_vec(edev, 6);
+
+  kap = 1.0 - lambda_->value(T) + lambda_->value(T) * de / eps0_;
+
+  return 0;
+}
+
+int WalkerKremplSwitchRule::dkappa(const double * const edot, double T,
+                                   double * const dkap)
+{
+  std::copy(edot, edot+6, dkap);
+  dev_vec(dkap);
+
+  if (norm2_vec(dkap, 6) == 0.0) {
+    for (size_t i = 0; i < 6; i++)
+      dkap[i] = 0.0;
+    return 0;
+  }
+
+  double fact = lambda_->value(T)  / eps0_ * std::sqrt(2.0/3.0) / norm2_vec(dkap, 6);
+
+  for (size_t i = 0; i < 6; i++)
+    dkap[i] *= fact;
+
+  return 0;
+}
+
+void WalkerKremplSwitchRule::override_guess(double * const x)
+{
+  flow_->override_guess(x);
+}
+
+SofteningModel::SofteningModel()
+{
+
+}
+
+std::string SofteningModel::type()
+{
+  return "SofteningModel";
+}
+
+std::unique_ptr<NEMLObject> SofteningModel::initialize(ParameterSet & params)
+{
+  return neml::make_unique<SofteningModel>(); 
+}
+
+ParameterSet SofteningModel::parameters()
+{
+  ParameterSet pset(SofteningModel::type());
+
+  return pset;
+}
+
+double SofteningModel::phi(double alpha, double T) const
+{
+  return 1.0;
+}
+
+double SofteningModel::dphi(double alpha, double T) const
+{
+  return 0.0;
+}
+
+WalkerSofteningModel::WalkerSofteningModel(std::shared_ptr<Interpolate> phi0,
+                                           std::shared_ptr<Interpolate> phi1) :
+    phi_0_(phi0), phi_1_(phi1)
+{
+
+}
+
+std::string WalkerSofteningModel::type()
+{
+  return "WalkerSofteningModel";
+}
+
+std::unique_ptr<NEMLObject> WalkerSofteningModel::initialize(ParameterSet & params)
+{
+  return neml::make_unique<WalkerSofteningModel>(
+      params.get_object_parameter<Interpolate>("phi_0"),
+      params.get_object_parameter<Interpolate>("phi_1")
+      ); 
+}
+
+ParameterSet WalkerSofteningModel::parameters()
+{
+  ParameterSet pset(WalkerSofteningModel::type());
+
+  pset.add_parameter<NEMLObject>("phi_0");
+  pset.add_parameter<NEMLObject>("phi_1");
+
+  return pset;
+}
+
+double WalkerSofteningModel::phi(double alpha, double T) const
+{
+  if (alpha <= 0.0) {
+    return 1.0;
+  }
+  else if (alpha < ainc_) {
+    return phi_0_->value(T) * std::pow(ainc_, phi_1_->value(T)) / ainc_ * alpha +
+        1.0;
+  }
+  else {
+    return 1.0 + phi_0_->value(T) * std::pow(alpha, phi_1_->value(T));
+  }
+}
+
+double WalkerSofteningModel::dphi(double alpha, double T) const
+{
+  if (alpha <= 0.0) {
+    return phi_0_->value(T) * std::pow(ainc_, phi_1_->value(T)) / ainc_;;
+  }
+  else if (alpha < ainc_) {
+    return phi_0_->value(T) * std::pow(ainc_, phi_1_->value(T)) / ainc_;
+  }
+  else {
+    return phi_1_->value(T) * phi_0_->value(T) * std::pow(alpha, phi_1_->value(T) - 1.0);
+  }
```

Comment:
I always try avoiding this style as some compiler my complain.
better doing something like
```
double a =0
if ...
  a = b
else 
  a = c
return a
```

Reply:
same above

Reply:
Yes, I need to be consistent.  Hopefully I caught them all now.